File: src/main/java/com/jaffa/rpc/lib/spring/ServerEndpoints.java
package com.jaffa.rpc.lib.spring;

import com.jaffa.rpc.lib.common.OptionConstants;
import lombok.Getter;

import java.util.Objects;

@Getter
public class ServerEndpoints {
    private final Class<?>[] endpoints;

    public ServerEndpoints(Class<?>... endpoints) {
        if (Objects.isNull(endpoints))
            throw new IllegalArgumentException(OptionConstants.ILLEGAL_ARGS_MESSAGE);
        this.endpoints = endpoints;
    }
}

File: src/main/kotlin/com/jaffa/rpc/lib/spring/ServerEndpoints.kt
package com.jaffa.rpc.lib.spring

import com.jaffa.rpc.lib.common.OptionConstants

class ServerEndpoints(vararg endpoints: Class<*>) {

    val endpoints: Array<Class<*>>

    init {
        require(true) { OptionConstants.ILLEGAL_ARGS_MESSAGE }
        this.endpoints = arrayOf(*endpoints)
    }
}